nginx.tmpl

```diff
@@ -0,0 +1,177 @@
+worker_processes	1;
+
+events {
+    worker_connections	1024;
+}
+
+{{$SSL := or (key "config/mantlui/ssl") (or (env "MANTLUI_SSL") "false") | parseBool}}
+{{$AUTH := or (key "config/mantl/auth") (or (env "MANTLUI_AUTH") "false") | parseBool}}
+{{$consulSSL := or (key "config/mantl/consul_ssl") (or (env "CONSUL_SSL") "false") | parseBool}}
+
+http {
+    include		mime.types;
+    default_type	application/octet-stream;
+
+    sendfile		on;
+    keepalive_timeout	65;
+
+    {{$mesos := service "leader.mesos"}}{{with $mesos}}
+    upstream mesos { {{range $mesos}}
+        server {{.Address}}:{{.Port}};{{end}}
+    } {{end}}
+
+    {{$marathon := service "marathon"}}{{with $marathon}}
+    upstream marathon { {{range $marathon}}
+        server {{.Address}}:{{.Port}};{{end}}
+    } {{end}}
+
+    {{$chronos := service "chronos"}}{{with $chronos}}
+    upstream chronos { {{range $chronos}}
+        server {{.Address}}:{{.Port}};{{end}}
+    } {{end}}
+
+    {{$consul := service "consul"}}{{with $consul}}
+    upstream consul { {{range $consul}}
+        server {{.Address}}:8500;{{end}}
+    } {{end}}
+
+    {{$mantlapi := service "mantl-api"}}{{with $mantlapi}}
+    upstream mantlapi { {{range $mantlapi}}
+        server {{.Address}}:{{.Port}};{{end}}
+    } {{end}}
+
+    {{$traefikAdmin := service "traefik-admin"}}{{with $traefikAdmin}}
+    upstream traefik-admin { {{range $traefikAdmin}}
+        server {{.Address}}:{{.Port}};{{end}}
+    } {{end}}
+
+    {{if $SSL}}
+    server {
+        listen 80;
+        return 301 https://$host$request_uri;
+    }
+    {{end}}
+
+    server {
+        listen {{if $SSL}}443 ssl{{else}}80{{end}};
+
+        {{if $SSL}}
+        ssl_certificate     {{or (env "CERTFILE") "/etc/mantlui/mantlui.cert"}};
+        ssl_certificate_key	{{or (env "KEYFILE")  "/etc/mantlui/mantlui.key"}};
+
+        ssl on;
+        ssl_session_cache	        builtin:1000 shared:SSL:10m;
+        ssl_protocols             TLSv1 TLSv1.1 TLSv1.2;
+        ssl_ciphers               {{or (key "config/mantlui/ssl_ciphers") "HIGH:!aNULL:!eNULL:!EXPORT:!CAMELLIA:!DES:!MD5:!PSK:!RC4"}};
+        ssl_prefer_server_ciphers on;
+
+        error_page 497 https://$host:$server_port$request_uri;
+        {{end}}
+
+        {{if $AUTH}}
+        auth_basic		on;
+        auth_basic_user_file	/etc/nginx/nginx-auth.conf;
+        {{end}}
+
+        location / {
+            proxy_connect_timeout	600;
+            proxy_send_timeout	600;
+            proxy_read_timeout	600;
+            send_timeout		600;
+
+            root /usr/share/nginx/html;
+            index index.html index.htm;
+            try_files $uri $uri/ =404;
+        }
+
+        {{with $mesos}}
+        location ~ /mesos/((master|metrics|monitor|files)\/.*) {
```

Comment:
@BrianHicks can you think of anything more clever to do here? The mesos api endpoints are not cleanly namespaced. The intention is for all mesos UI requests to hit the static files embedded in the container (`location /mesos` below) but all API requests to hit the upstream mesos master.